Verona/Verona.swift

```diff
@@ -102,17 +102,73 @@ public enum NSAttributedStringAttribute {
 
 public class NSAttributedStringAttributeBuilder {
     
-    private var attributesDictionary: [String : AnyObject] = [ : ]
+    private var clearAttributesOnNextString: Bool = false
+    
+    private var currentString: String = ""
+    private var strings = [String]()
+    
+    var currentAttributes: [String : AnyObject] = [ : ]
+    var attributes = [[String: AnyObject]]()
     
     public init() { }
     
-    public func setAttribute(attribute: NSAttributedStringAttribute) {
+    // Builder functions
+    
+    public func setAttribute(attribute: NSAttributedStringAttribute) throws {
         let keyValuePair = attribute.attributedStringKeyValuePair
-        attributesDictionary.updateValue(keyValuePair.value, forKey: keyValuePair.key)
+        currentAttributes.updateValue(keyValuePair.value, forKey: keyValuePair.key)
     }
     
-    public func attributedStringAttributesDictionary() -> [String: AnyObject]? {
-        return attributesDictionary
+    public func clearAttributes() {
+        currentAttributes.removeAll()
+    }
+    
+    public func addAttributes(attributes: [String: AnyObject]) {
```

Comment:
Potentially we should rename this to `updateCurrentAttributes` as it will replace existing attributes, and also only applies to the current `nextString`